### PR TITLE
Fix for Issue #38

### PR DESCRIPTION
**Pull request by Bobnix**
I added and configured a pom file for use with maven. As long as maven is installed on the host machine, the project can be built with mvn clean install.

### CODE REVIEW

**Comment by Bobnix** [comment]
I added a travis ci config file just because it was a trivial amount of effort. It does not force us to use Travis CI, the option is simply there. When we have tests, it would be nice to have it automatically build and test.

**Comment by ceiroa** [comment]
We have an existing Jenkins instance where we run all our Foundation Products Team builds. It might be nice to get it plugged in there.

**Comment by Bobnix** [comment]
Tying it to Jenkins would be great. I have a bit of a crush on Jenkins/Hudson as far as build systems go.

File: .classpath [hunk 1]
```diff
@@ -1,6 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.6"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry including="**/*.java" kind="src" output="target/classes" path="src">
```
Comment (davidhabib) [inline .classpath]: So I am not actually familiar with the contents of this file, as it was maintained by Eclipse.  It seems like you are changing the output location of compiled classes.  Will this still work with Eclipse?

File: .classpath [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+		<attributes>
+			<attribute name="optional" value="true"/>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/J2SE-1.5">
```
Comment (davidhabib) [inline .classpath]: looks like the version of Java is being changed by this line.  what was the motivation for changing?

File: .classpath [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+		<attributes>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.m2e.MAVEN2_CLASSPATH_CONTAINER">
+		<attributes>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="output" path="target/classes"/>
```
Comment (davidhabib) [inline .classpath]: another change from what Eclipse defaultly does.  will this change prevent Eclipse from compiling the project?

File: .classpath (hunk 1 above)
Comment (Bobnix) [inline .classpath]: This was altered by my eclipse actually, I am not 100% sure what happened

**Comment by Bobnix** [comment]
What was did your build process look like before? What were the steps to do it?

**Comment by davidhabib** [comment]
with the ApexDoc project selected in Eclipse, choose File, Export menu command.  in the Export dialog, choose Java > Runnable JAR file.  On the Runnable JAR File Export page, choose Extract required libraries into generated JAR.  (I might have chosen the Package option instead; can't remember).

**Comment by Bobnix** [comment]
You were 100% right, while this setup worked fine for the maven flow, the export did not. I changed the class filter to grab all the files (including the resources) into the target jar. I also bumped the JRE version while I was at it though I am not sure if you want it set higher.

**Comment by davidhabib** [comment]
OK, I will merge this and test out to make sure it didn't negatively affect my build in Eclipse.

**Comment by davidhabib** [comment]
building the jar with Eclipse still works fine, so we are good.

**Comment by Bobnix** [comment]
*Phew*, I cant tell you how nervous I was about that one. Eclipse, for me at least, has always been a wonderfully fickle tool. Often times code that builds for one person would flat out refuse due to an otherwise innocuous settings problem. I am happy to see this work.

**Pull request merged**